x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/result_settings_table/result_settings_field_number.test.tsx

```diff
@@ -0,0 +1,108 @@
+    fieldSizeProperty: 'rawSize',
+    updateAction: jest.fn(),
+    clearAction: jest.fn(),
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("is rendered with it's value set from [fieldSizeProperty] in fieldSettings", () => {
+    const wrapper = shallow(<ResultSettingsFieldNumber {...props} />);
+    expect(wrapper.find(EuiFieldNumber).prop('value')).toEqual(29);
+  });
+
+  it('is has no value if [fieldSizeProperty] in fieldSettings has no value', () => {
+    const wrapper = shallow(
+      <ResultSettingsFieldNumber
+        {...{
+          ...props,
+          fieldSettings: {
+            raw: true,
+            snippet: false,
+          },
+        }}
+      />
+    );
+    expect(wrapper.find(EuiFieldNumber).prop('value')).toEqual('');
+  });
+
+  it('is disabled if the [fieldEnabledProperty] in fieldSettings is false', () => {
+    const wrapper = shallow(
+      <ResultSettingsFieldNumber
+        {...{
+          ...props,
+          fieldSettings: {
+            raw: false,
+            snippet: true,
+          },
+        }}
+      />
+    );
+    expect(wrapper.find(EuiFieldNumber).prop('disabled')).toEqual(true);
+  });
+
+  it('will call updateAction when the value is changed', () => {
+    const wrapper = shallow(<ResultSettingsFieldNumber {...props} />);
+    wrapper.simulate('change', { target: { value: '21' } });
+    expect(props.updateAction).toHaveBeenCalledWith('foo', 21);
+  });
+
+  it('will call clearAction when the value is changed', () => {
+    const wrapper = shallow(<ResultSettingsFieldNumber {...props} />);
+    wrapper.simulate('change', { target: { value: '' } });
+    expect(props.clearAction).toHaveBeenCalledWith('foo');
+  });
+
+  it('will call updateAction on blur', () => {
+    const wrapper = shallow(<ResultSettingsFieldNumber {...props} />);
+    wrapper.simulate('blur', { target: { value: '21' } });
+    expect(props.updateAction).toHaveBeenCalledWith('foo', 21);
+  });
+
+  it('will call updateAction on blur using the minimum possible value if the current value is something other than a number', () => {
+    const wrapper = shallow(<ResultSettingsFieldNumber {...props} />);
+    wrapper.simulate('blur', { target: { value: '' } });
+    expect(props.updateAction).toHaveBeenCalledWith('foo', 20);
+  });
+
+  it('will call updateAction on blur using the minimum possible value if the value is something lower than the minimum', () => {
+    const wrapper = shallow(<ResultSettingsFieldNumber {...props} />);
+    wrapper.simulate('blur', { target: { value: 5 } });
+    expect(props.updateAction).toHaveBeenCalledWith('foo', 20);
+  });
+
+  it('will call updateAction on blur using the maximum possible value if the value is something above than the maximum', () => {
+    const wrapper = shallow(<ResultSettingsFieldNumber {...props} />);
+    wrapper.simulate('blur', { target: { value: 2000 } });
+    expect(props.updateAction).toHaveBeenCalledWith('foo', 1000);
+  });
+});
```

Comment:
I'm just here to say I SUPER appreciate this having been pulled out to its own component, and also these unit tests make both the normal flow & edge cases super clear! 😍